### DISCUSSION TITLE
Best way to create an .app (apart from symlinking)

### DISCUSSION THREAD

**Discussion by memeplex** [d12frosted/homebrew-emacs-plus#402] [Q&A] [answered]
Hi all,

the official recommendation is to symlink:

```
ln -s /usr/local/opt/emacs-plus@28/Emacs.app /Applications
```

so my question here may seem redundant. But the reason I'm asking is that a symlink won't appear in Spotlight.

Other alternatives I explored are:

* Create an Alias. Better, but it still appears with low priority at the bottom of Spotlight, inside Other category.
* Create a shell script app with Automator. Puts an absurd gear icon on the systray.
* Create an app with Automator. Better, but the gear is there during launch.
* Copy the app package from /usr/local/opt to /Applications. Some space duplication, extra manual update step.
* Move the app package instead of copying it. No duplication, still extra step, don't know how well it plays with brew upgrades.

What do you think? Is it worth providing a cask install?

Best regards,
Carlos

**Comment by d12frosted** [comment, marked as answer]
> Create an Alias. Better, but it still appears with low priority at the bottom of Spotlight, inside Other category.

This should be the same as creating alias. Or am I missing something? 

> What do you think? Is it worth providing a cask install?

See https://github.com/d12frosted/homebrew-emacs-plus/issues/41

In short, I am not planning to invest my time in `brew cask`. So unless someone contributes, it is not going to happen :) 

There are several reasons for me not willing to work on that: (1) that's not how I start Emacs anyways, (2) over years of maintaining `emacs-plus` I become too tired of `brew` ecosystem and (3) `brew` doesn't like when someone provides installation options in their formulas (if cask is akin to bottles in terms of options support, then cask is useless for `emacs-plus`).

**Comment by memeplex** [comment]
Create an Alias. Better, but it still appears with low priority at the
> bottom of Spotlight, inside Other category.
>
> This should be the same as creating alias. Or am I missing something?
>

Alias is a macOS thing, more like a direct access in Windows. Spotlight
seems to like aliases more than symlinks but still less than proper app
packages. Symlinks won't even appear while aliases are shown at the bottom.

What do you think? Is it worth providing a cask install?
>
> See #41 <https://github.com/d12frosted/homebrew-emacs-plus/issues/41>
>
> In short, I am not planning to invest my time in brew cask. So unless
> someone contributes, it is not going to happen :)
>

Ok, no worries, no big deal. Thank you for taking the time to answer.

>
>

**Comment by d12frosted** [comment]
> Alias is a macOS thing

Indeed, they are different. For some reason I thought they are both the same 😅 

> What do you think? Is it worth providing a cask install?
> Ok, no worries, no big deal. Thank you for taking the time to answer.

Don't want to confuse nor discourage anyone 😸 Just wanted to explain why I am not willing to do that by myself, but I will gladly accept contributions. And if someone wants to bite this bullet, just let me know, so I can assist and provide all the information I have.

**Reply by memeplex** [reply]
Here is a very interesting albeit inconclusive discussion about these issues https://github.com/Homebrew/legacy-homebrew/issues/4635. It seems like I had more or less exhausted the possibilities above. There is no clear cut solution and there is this gulf between apps that live in /usr/local and apps that are binary and can be installed in /Applications, but nothing that can be built from sources, put a part in /usr/local and still expose the GUI in /Applications. Not the best state of affairs but that's the situation. Moreover, it seems that Spotlight does read symlinks, but that /usr is hidden (and /opt isn't) and that's the problem, but more recent version of macOS won't let you change the visibility of /usr.
